### PR TITLE
Add support for capturing additional upload metadata in Participants database

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Update Upload table schema and corresponding c# data access objects to support capturing additional Upload metadata.

## Why?

NAC-915, NAC-917

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [x] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: participants/src/Piipan.Participants/Piipan.Participants.Api/Models/IUpload.cs [hunk 1]
```diff
@@ -5,7 +5,13 @@ namespace Piipan.Participants.Api.Models
...
+        string UploadIdentifier { get; set; }
         DateTime CreatedAt { get; set; }
         string Publisher { get; set; }
+
+        long? ParticipantsUploaded { get; set; }
+        string ErrorMessage { get; set; }
+        DateTime? CompletedAt { get; set; }
+        string Status { get; set; }
     }
 }
```
Comment (ccaprio97) [inline participants/src/Piipan.Participants/Piipan.Participants.Api/Models/IUpload.cs:17]: for the future. Harvey said we should be having new lines at the end of files. Not sure why, i didnt know it either

File: participants/src/Piipan.Participants/Piipan.Participants.Core/Models/UploadDbo.cs [hunk 2]
```diff
@@ -6,7 +6,12 @@ namespace Piipan.Participants.Core.Models
     public class UploadDbo : IUpload
     {
         public Int64 Id { get; set; }
+        public string UploadIdentifier { get; set; }
         public DateTime CreatedAt { get; set; }
         public string Publisher { get; set; }
+        public long? ParticipantsUploaded { get; set; }
```
Comment (ccaprio97) [inline participants/src/Piipan.Participants/Piipan.Participants.Core/Models/UploadDbo.cs:12]: can you upload with 0 participants?

File: participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantService.cs [hunk 3]
```diff
@@ -111,7 +114,8 @@ public async Task AddParticipants(IEnumerable<IParticipant> participants, string
             }
             catch (Exception ex)
             {
-                await _uploadDao.UpdateUploadStatus(upload, UploadStatuses.FAILED.ToString());
+                upload.Status = UploadStatuses.FAILED.ToString();
```
Comment (ccaprio97) [inline participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantService.cs:117]: is the logger still here?

**Review by ccaprio97 — APPROVED** [review]
had some comments but none were stoppages. LGTM

File: participants/src/Piipan.Participants/Piipan.Participants.Core/Models/UploadDbo.cs (hunk 2 above)
Comment (bgfoshay) [reply participants/src/Piipan.Participants/Piipan.Participants.Core/Models/UploadDbo.cs:12]: When the record first gets created (i.e. the Upload is processing) the number of participants would be null. At the point it completes processing we have a count of the number of participants that got uploaded. So when we update the record, we can set the number as well as update the status to "Complete"

**Pull request merged**